core/arch/arm/kernel/arch_scall.c

```diff
@@ -94,9 +95,15 @@ static void save_panic_regs_a32_ta(struct thread_specific_data *tsd,
 static void save_panic_regs_a64_ta(struct thread_specific_data *tsd,
 				   uint64_t *pushed)
 {
+	uint64_t x29 = 0;
+	uint64_t elr = 0;
+
+	GET_USER(x29, &pushed[0]);
+	GET_USER(elr, &pushed[1]);
```

Comment:
check return value from `GET_USER()`. If I understand correctly the implementation, you can set `tsd->abort_regs.x29` and `tsd->abort_regs.elr` to 0 upon failure.

@jenswi-linaro, @jforissier: a expert advise is welcome here. Is user land x1 register always trustfull in this case?

Reply:
added some return value checks